src/tools/cephfs/DataScan.h

```diff
@@ -220,7 +220,7 @@ class MetadataDriver : public RecoveryDriver, public MetadataTool
 
     int inject_linkage(
         inodeno_t dir_ino, const std::string &dname,
-        const frag_t fragment, const InodeStore &inode, snapid_t dnfirst=CEPH_NOSNAP);
+        const frag_t fragment, const InodeStore &inode, snapid_t dnfirst=2);
```

Comment:
'cephfs-data-scan scan_links' should fix this.

Reply:
Tried, it is still the 'head' and it crashes.

Reply:
Sorry, it works, it is 2.